zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -716,6 +719,13 @@ private void pRequest2TxnCreate(int type, Request request, Record record, boolea
             ephemeralOwner = EphemeralType.TTL.toEphemeralOwner(ttl);
         } else if (createMode.isEphemeral()) {
             ephemeralOwner = request.sessionId;
+
+            int count = zks.getZKDatabase().getDataTree().getEphemerals(ephemeralOwner).size();
```

Comment:
is this thread safe?

Reply:
The underlying map is a ConcurrentHashMap so the operations are threadsafe. But there is no guarantee that our get request - size() - won't overlap with a put and/or remove operation. 

The currnet proposed design for this accepts that there will be a degree of inaccuracy in enforcing the quota due to parallel requests and the time for the request pipeline to complete.